### PR TITLE
Move injector to dependency resolver

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

This is adopt PR to https://github.com/yiisoft/factory/pull/123 and instead of https://github.com/yiisoft/di/pull/222

### CODE REVIEW

File: src/DependencyResolver.php [hunk 1]
```diff
@@ -56,8 +58,21 @@ public function resolve(string $id)
...
     {
         return false;
     }
+
+    private function getInjector(): Injector
+    {
+        if ($this->injector === null) {
+            $this->injector = new Injector($this->container);
+        }
+        return $this->injector;
```
Comment (xepozz) [inline src/DependencyResolver.php:76]:
```suggestion
        return $this->injector ??= new Injector($this->container);
```

File: src/DependencyResolver.php [hunk 2]
```diff
@@ -56,8 +58,21 @@ public function resolve(string $id)
...
+    {
+        return $this->getInjector()->invoke($callable);
+    }
+
     public function shouldCloneOnResolve(): bool
     {
         return false;
     }
+
+    private function getInjector(): Injector
```
Comment (xepozz) [inline src/DependencyResolver.php:71]: What about to move the initialization into __construct method?

File: src/DependencyResolver.php (hunk 1 above)
Comment (vjik) [reply src/DependencyResolver.php]: Done.

File: src/DependencyResolver.php (hunk 2 above)
Comment (vjik) [reply src/DependencyResolver.php:71]: injector may not be needed

**Review by samdark — APPROVED** [review]

**Pull request merged**